modules/tensor_mechanics/src/materials/DiscreteRRIsotropicPlasticity.C

```diff
@@ -0,0 +1,150 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "DiscreteRRIsotropicPlasticity.h"
+
+template<>
+InputParameters validParams<DiscreteRRIsotropicPlasticity>()
+{
+  InputParameters params = validParams<DiscreteRadialReturnStressIncrement>();
+
+  // Linear strain hardening parameters
+  params.addParam<Real>("yield_stress", "The point at which plastic strain begins accumulating");
+  params.addParam<FunctionName>("yield_stress_function", "Yield stress as a function of temperature");
+  params.addParam<Real>("hardening_constant", "Hardening slope");
+  params.addParam<FunctionName>("hardening_function", "True stress as a function of plastic strain");
+  params.addCoupledVar("temp", "Coupled Temperature");
+
+  return params;
+}
+
+
+DiscreteRRIsotropicPlasticity::DiscreteRRIsotropicPlasticity( const InputParameters & parameters)
+  :DiscreteRadialReturnStressIncrement(parameters),
+   _yield_stress_function(isParamValid("yield_stress_function") ? &getFunction("yield_stress_function") : NULL),
+   _yield_stress(isParamValid("yield_stress") ? getParam<Real>("yield_stress") : 0),
+   _hardening_constant(isParamValid("hardening_constant") ? getParam<Real>("hardening_constant") : 0),
+   _hardening_function(isParamValid("hardening_function") ? dynamic_cast<PiecewiseLinear*>(&getFunction("hardening_function")) : NULL),
+
+   _plastic_strain(declareProperty<RankTwoTensor>("plastic_strain")),
+   _plastic_strain_old(declarePropertyOld<RankTwoTensor>("plastic_strain")),
+   _scalar_plastic_strain(declareProperty<Real>("scalar_plastic_strain")),
+   _scalar_plastic_strain_old(declarePropertyOld<Real>("scalar_plastic_strain")),
+
+   _hardening_variable(declareProperty<Real>("hardening_variable")),
+   _hardening_variable_old(declarePropertyOld<Real>("hardening_variable")),
+   _shear_modulus(declareProperty<Real>("shear_modulus")),
+   _temperature(isCoupled("temp") ? coupledValue("temp") : _zero)
```

Comment:
Initialization lists should be indented 4 spaces for MOOSE code. Also, I think we typically put the `:` on the first line, I don't know if this is official or not.